src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Summarize.txt

```diff
@@ -0,0 +1,408 @@
+﻿#SETUP: SupportColumnNamesAsIdentifiers,ConsistentOneColumnTableResult,PowerFxV1CompatibilityRules
+
+// Average
+// Max
+// Min
+// StdevP
+// VarP
+// Count
+// CountA
+// CountRows
+// CountIf
+// Invalid functions
+
+
+// === Basic grouping with aggregates =================================================
+
+>> Summarize( 
+  Table({Supplier:"Contoso",  Fruit:"Grapes",  Price:220, Purchase:Date(2015,10,1), Tags: ["Red","Seedless"]},
+  {Supplier:"Fabrikam", Fruit:"Lemons",  Price:31,  Purchase:Date(2015,10,1), Tags: ["Colombia"]},
+  {Supplier:"Contoso",  Fruit:"Lemons",  Price:29,  Purchase:Date(2015,10,2), Tags: ["Peru"]},
+  {Supplier:"Contoso",  Fruit:"Grapes",  Price:210, Purchase:Date(2015,10,2), Tags: ["Green","Seedless"]},
+  {Supplier:"Fabrikam", Fruit:"Lemons",  Price:30,  Purchase:Date(2015,10,3), Tags: ["Mexixo","Seedless"]},
+  {Supplier:"Contoso",  Fruit:"Bananas", Price:12,  Purchase:Date(2015,10,3), Tags: ["Mexico"]}),
+  Supplier, Fruit, CountRows( ThisGroup ) As Count )
+Table({Count:2,Fruit:"Grapes",Supplier:"Contoso"},{Count:2,Fruit:"Lemons",Supplier:"Fabrikam"},{Count:1,Fruit:"Lemons",Supplier:"Contoso"},{Count:1,Fruit:"Bananas",Supplier:"Contoso"})
+
+// Aggregates last
+>> Summarize(
+  Table({Supplier:"Contoso",  Fruit:"Grapes",  Price:220, Purchase:Date(2015,10,1), Tags: ["Red","Seedless"]},
+  {Supplier:"Fabrikam", Fruit:"Lemons",  Price:31,  Purchase:Date(2015,10,1), Tags: ["Colombia"]},
+  {Supplier:"Contoso",  Fruit:"Lemons",  Price:29,  Purchase:Date(2015,10,2), Tags: ["Peru"]},
+  {Supplier:"Contoso",  Fruit:"Grapes",  Price:210, Purchase:Date(2015,10,2), Tags: ["Green","Seedless"]},
+  {Supplier:"Fabrikam", Fruit:"Lemons",  Price:30,  Purchase:Date(2015,10,3), Tags: ["Mexixo","Seedless"]},
+  {Supplier:"Contoso",  Fruit:"Bananas", Price:12,  Purchase:Date(2015,10,3), Tags: ["Mexico"]}),
+  Supplier, Fruit, Average( ThisGroup, Price ) As Avg_Price )
+Table({Avg_Price:215,Fruit:"Grapes",Supplier:"Contoso"},{Avg_Price:30.5,Fruit:"Lemons",Supplier:"Fabrikam"},{Avg_Price:29,Fruit:"Lemons",Supplier:"Contoso"},{Avg_Price:12,Fruit:"Bananas",Supplier:"Contoso"})
```

Comment:
>{Avg_Price:215,Fruit:"Grapes",Supplier:"Contoso"} [](http://example.com/codeflow?start=6&length=49)

Should we have same ordering as defined by end user in function? or we consciously decided to reverse it?

Reply:
Dont think order is important. Nothing has been changed to intentionally reverse.

Reply:
I feel we should keep the ordering as how user defined it,
Similar to SQL,
`select f1, f2, Count(*) as f3 from table`
Will always have deterministic ordering as per how user ordered.

Reply:
Let me rephrase myself: we print the expression in a deterministic order.
![image](https://github.com/microsoft/Power-Fx/assets/19541904/baae2a4f-e994-4759-8e59-c88f2ed75ddc)

What I meant was this is not something I changed in the Summarize function.

Reply:
Ahh! that makes sense!